### PR TITLE
Seansaoirse api contact

### PR DESCRIPTION
**Pull request by shonsirsha**
This PR is a WIP (work in progress). Here are the tasks that are done, and not yet done:

- Using one model for creating a new contact (for both sending to API  and displaying it on the frontend) ✅
- Creating a new contact API integration ✅

For the next tasks, please please read this docs. It's not really that long: https://github.com/shonsirsha/chimp_backend#authentication-flow 

 So we have the same understanding on the topic :-) 

1 Checking if request to the API has failed and if token has expired ✅
2 Getting a new token if the token used was expired (refreshing token mechanism) ✅  
3 Setting the newly acquired token (the new access token) to the `@State` token ✅  
4 Setting the newly acquired token (the new access token) to the `token` attribute of the `AuthDetail` CoreData entity 😞  
5 Hitting the create contact API again after the token has been renewed. ❌  - This task is basically blocked by the task before this. 

So basically, the API integration is almost done. As we can see, there's two major tasks left. 

@mgravemeyer I need your help on 4, which is: updating CoreData values for  AuthDetail entity for the `token` attribute.

### CODE REVIEW

**Comment by shonsirsha** [comment]
@mgravemeyer  **creating a new contact in the DB via the API** and the access token refreshing mechanism are all working. 

As we discussed before, this refreshing mechanism works behind the scene and user won't notice it (getting new access token once the current one is expired, which is every 15mins)
